ViewModel/ExpressCheckoutButton.php

```diff
@@ -76,15 +76,19 @@ public function __construct(
      */
     public function shouldRender(): bool
     {
-        return $this->isExpressCheckoutActive();
+        return $this->isExpressCheckoutDisabled();
     }
 
     /**
      * @return bool
      */
-    public function isExpressCheckoutActive(): bool
+    public function isExpressCheckoutDisabled(): bool
```

Comment:
Using the `!()` statment means you do a negative check on whatever is between the parenthesis.

In this case we check:

- Is the module disabled (`!$this->config->isActive()`)
- Is the express checkout disabled (`!$this->config->isExpressCheckoutActive()`)
- Is the current current code not the allowed currency code (`$this->storeManager->getStore()->getCurrentCurrencyCode() != self::AMWAL_CURRENCY`)
- Is the Quote total exactly 0 (`$this->checkoutSessionFactory->create()->getQuote()->getGrandTotal() == 0`)

If any of these statements are true, it means the express checkout is disabled, so when our check simplified is:
`!$isCheckoutDisabled` 

Meaning the naming of the function would more logically be `isExpressCheckoutActive` or `isExpressCheckoutEnabled`